### PR TITLE
Replace MICE with IterativeImputer + misc fixes to warrant version bump

### PR DESCRIPTION
**Pull request by sergeyf**
During the long development of `MICE` in `sklearn`, I learned that my approach to multiple imputation was not right. You're not supposed to average the last `n_imputations`. What you're supposed to do is create `n_imputations` separate imputers, and then run them all entirely in parallel. The only difference is each one has to have its own different random seed. 

So, what I've done in this PR is removed `MICE` and replace it with `IterativeImputer`, which is the current version in the `sklearn` PR. However, the `sklearn` version of `IterativeImputer` is slated to be in `sklearn 0.21` which is 2 versions ahead of current `0.19.1`, so it could be up to a year. 

As such, I wanted to add it to `fancyimpute` and provide what I consider to be the correct version.

@iskandr please take a look. I think it's pretty straightforward. The tests have an example of how to use `IterativeImputer` as MICE.

### CODE REVIEW

**Comment by sergeyf** [comment]
Also there is a lot of extra stuff in `iterative_imputer.py`. This is because `IterativeImputer` was written to work with `sklearn 0.20`, but that's not out yet. As soon as it's pip installable (a few weeks I bet), I can make another PR to remove all the extras (like `_SimpleImputer`) and up the requirement.

File: README.md [hunk 1]
```diff
@@ -56,3 +64,52 @@ on features for which two rows both have observed data.
...
+to obtain understanding of how analytic results may differ as a consequence
+of the inherent uncertainty caused by the missing values. The above practice
+is called multiple imputation.
+
+Our implementation of `IterativeImputer` was inspired by the R MICE
+package (Multivariate Imputation by Chained Equations) [1], but differs from
+it by returning a single imputation instead of multiple imputations.  However,
+IterativeImputer` can also be used for multiple imputations by applying
+it repeatedly to the same dataset with different random seeds when
+``sample_posterior=True``. A quick example:
```
Comment (iskandr) [inline README.md]: Huh! I didn't realize how much we're diverging from what MICE means. Maybe we should eventually add a MICE wrapper which runs multiple random seed IterativeImputer instances, passes their results to a user-supplied function, and averages the analysis results? 

Even if that's desirable, definitely for a future PR.

File: fancyimpute/solver.py [hunk 2]
```diff
@@ -69,7 +67,7 @@ def _fill_columns_with_fn(self, X, missing_mask, col_fn):
                 continue
             col_data = X[:, col_idx]
             fill_values = col_fn(col_data)
-            if np.isnan(fill_values):
+            if np.all(np.isnan(fill_values)):
```
Comment (iskandr) [inline fancyimpute/solver.py]: Never noticed that bug, good catch.

File: README.md (hunk 1 above)
Comment (sergeyf) [reply README.md]: Ya I was thinking the same, but I'm not sure if it's that simple. I think the imputation algo has to be sufficiently random in some sense I don't fully get.

**Pull request merged**